syntax = "proto3";

package car_ipc;

// The greeter service definition.
service Greeter {
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
    string name = 1;
}

// The response message containing the greetings.
message HelloReply {
    string message = 1;
}

service MovementClient {
    rpc MoveForward (MoveRequest) returns (MoveReply) {}
    rpc MoveBackward (MoveRequest) returns (MoveReply) {}
    rpc TurnLeft (TurnRequest) returns (TurnReply) {}
    rpc TurnRight (TurnRequest) returns (TurnReply) {}
}

message MoveRequest {
    double distance = 1; // in meters
}

message MoveReply {
    string status = 1;
}

message TurnRequest {
    double angle = 1; // in degrees
}

message TurnReply {
    string status = 1;
}